src/arrangement/core.cljc

```diff
@@ -78,10 +90,11 @@
         (compare-seqs rank a b)
 
         :else
-        (let [class-diff (compare (.getName (class a))
-                                  (.getName (class b)))]
+        (let [class-diff (compare (type-name a)
+                                  (type-name b))]
           (if (zero? class-diff)
-            (if (instance? java.lang.Comparable a)
-              (compare a b)
-              (compare (pr-str a) (pr-str b)))
+            #?(:clj (if (instance? java.lang.Comparable a)
+                      (compare a b)
+                      (compare (pr-str a) (pr-str b)))
+               :cljs (compare a b))
```

Comment:
ClojureScript's `compare` is a bit more friendly than Clojure's, so we don't actually need to ensure it satisfies IComparable:

```
cljs.core/compare
([x y])
  Comparator. Returns a negative number, zero, or a positive number
  when x is logically 'less than', 'equal to', or 'greater than'
  y. Uses IComparable if available and google.array.defaultCompare for objects
 of the same type and special-cases nil to be less than any other object.```